### ISSUE TITLE
Current support for rich text cells

### ISSUE THREAD

**Issue by buffaybu** [dtjohnson/xlsx-populate#34]
Is rich text support coming soon?

If not, maybe it's useful that `aRichTextCell.value()` returns the `<si>` node instead of only its first child ([SharedString.js#L89](https://github.com/dtjohnson/xlsx-populate/blob/master/lib/SharedStrings.js#L89)) so that developers can somehow extract the string on their own.

**Comment by dtjohnson** [comment]
It should already work now for extracting rich text as you suggest. It just returns an XMLish object though. If you find that is not the case, could you please share some sample code and a workbook that exhibits the issue?

Setting rich text isn't currently supported. It is on the roadmap to support at some point in the future.

**Comment by buffaybu** [comment]
[REDBLACK.xlsx](https://github.com/dtjohnson/xlsx-populate/files/915002/REDBLACK.xlsx)
Cell A1 contains string 'REDBLACK'.

When parsing:
```
require('xlsx-populate').fromFileAsync("./ad.xlsx").then(workbook => {
  const value = workbook.sheet("Sheet1").cell("A1").value();
});
```

`value` is:
```
{
  name: 'r',
  attributes: {},
  children: [
    {
      name: 'rPr',
      children: [{ name: color, attributes: { rgb: 'FFFF0000' }, ... ],
      ...
    },
    {
      name: 't',
      attributes: {},
      children: ['RED']
    }
  ]
}
```
With this object I can never know 'BLACK'. What is expected is:
```
{
  name: 'si',
  children: [
    {
      name: 'r',
      attributes: {},
      children: [
        {
          name: 'rPr',
          children: [{ name: color, attributes: { rgb: 'FFFF0000' }, ... ],
          ...
        },
        {
          name: 't',
          attributes: {},
          children: ['RED']
        }
      ]
    },
    {
      name: 'r',
      attributes: {},
      children: [
        {
          name: 'rPr',
          children: [{ name: color, attributes: { rgb: 'FF000000' }, ... ],
          ...
        },
        {
          name: 't',
          attributes: {},
          children: ['BLACK']
        }
      ]
    }
  ]
}
```

In [SharedStrings.js#L89](https://github.com/dtjohnson/xlsx-populate/blob/master/lib/SharedStrings.js#L89), the item pushed into `_stringArray` is the first child of `<si>`, as it is supposed to be `<t>` in the first `if` branch, but it is not `<t>`, meaning that this cell might be a rich text cell or something not supported, so the pushed item should be the `<si>` node.

**Comment by dtjohnson** [comment]
Got it. Yep, that's a bug. I just published v1.5.5 which now returns the full array of children in rich text nodes instead of just the first one.

Long term the plan is to introduce two new classes: `RichText` and `RichTextRun`. The `RichTextRun` would be for each `r` node and the `RichText` would be for the collection. `RichTextRun` would have a `text` method and a `style` method similar to `Cell.style`. I'd also like a `text` getter method on `RichText`, which would give you the concatenation of all of the texts. It might be nice to have a `RichText.style` method as well like `Range.style`. When you get the value of a rich text cell it would return an instance of `RichText`. It would have to function like `Cell.style` where it creates a new copy per cell to prevent modification of all usages. I need to think about the easiest way to set a rich text value in a cell though.

**Comment by buffaybu** [comment]
Great! And thanks for the sharing.
